### DISCUSSION TITLE
Dataswap project(About Discussion on Trustless Notary)

### DISCUSSION THREAD

**Discussion by Kennynauh** [filecoin-project/community#660] [Project Showcase]
Hello everyone,

Dataswap is blockchain Layer 2 project built on IPFS and Filecoin. As a decentralized open data exchange platform, its objective is to aggregate open datasets from various regions and industries worldwide, enabling the permanent storage of valuable human-generated data.

About Discussion on Trustless Notary(https://docs.google.com/document/d/1KLR6nZ8ic4ARj3J46XsxSE_b1RpDP_z3JBKL4alHGGw/edit?pli=1) and Trustless Notary Design Space + Guidelines(https://medium.com/filecoin-plus/ideation-trustless-notary-design-space-guidelines-bc21f6d9d5f2), Dataswap(https://github.com/dataswap/specs/tree/main/systems#22-trustless-notary)  has delved deeper into its implementation within this particular area.
- Dataswap introduces the "Dataset Consistency Algorithm(https://github.com/dataswap/specs/blob/main/algorithms/README.md#2-dataset-consistency-algorithm)" for the first time, achieving verifiability in data storage to ensure the loading of accurate data.
- Automated and decentralized management and allocation of datacap at a finer granularity.
- Decoupling of datacap and notarized signatures to prevent malicious activities.

Storage providers pre-stake funds into the Dataswap Trustless Notary contract to participate in dataset auctions and secure validated data storage transactions.

The Dataswap project is open-source, and we warmly welcome  all community members to participate in contributing, testing, and discussing.

GitHub links:
- specs: https://github.com/dataswap/specs
- contract: https://github.com/dataswap/core
- algorithm: https://github.com/dataswap/go-metadata

**Comment by The-Wayvy** [comment]
Hey Kenny

I have 2 questions about your design.

1.) Can anyone mint DataCap?
2.) Can anyone participate in the audit process?

**Reply by waynewyang** [reply]
1 Minting datacap still maintains the current design, minted by layer1 (filecoin layer1), and dataswap storage is used as trustless notary need datacap from layer1.
Users only need to apply for datacap at layer2 (dataswap), not layer1.
But before the application, dataswap has guaranteed that this part of datacap can only be used for certified data.
2 The audit is divided into two parts, MDA and MDP, but I think the specific auditors can all retain the current notary at the beginning,
The role of MDA is to review whether the summary data of the dataset is valid and useful. In the later stage, MDA can review by industry to ensure that the data entering the filecon network is useful data.
The audit of MDP can actually be called verification, because this part is aimed at the real data of the actual data set. In this part, the reviewer cannot be fake, because dataswap will verify the results of the review on the chain, only the verification Passed can be regarded as a valid audit.

@Kennynauh 
 see if i explain clearly，If not, please help to add more comprehensive，
and relay here
,and updata our specs.

**Reply by Kennynauh** [reply]
Here are some additions
I will update the specs for better understanding

1 Datacap does not necessitate permission. This is because the allocation process is separate from the data set auditing within dataswap. The dataswap data set consistency algorithm ensuring the right data is onboarded. Governance operates at the level of each individual deal. Once data successfully passes the dataswap review procedure, any Storage Provider (SP) can carry out datacap allocation without requiring explicit authorization. In instances where datacap is illicitly employed, dataswap promptly monitors and halts the datacap allocation. 
2 The audit is divided into two parts. MDA is responsible for the metadata audit, that is, the audit of the basic characteristics of the data industry attributes. It is recommended that the existing notary team, DA, anyone can apply, because the certification and verification of the data set can be software Realized, anyone can verify it through the tool
